cmake/HalideGeneratorHelpers.cmake

```diff
@@ -680,3 +668,20 @@ function(_Halide_target_export_single_symbol TARGET SYMBOL)
         GNU_LD "${CMAKE_CURRENT_BINARY_DIR}/${TARGET}.ldscript"
     )
 endfunction()
+
+function(_Halide_gengen_ensure)
+    # Create a Generator that is GenGen.cpp and nothing else; all it can do is generate a runtime.
+    if (NOT TARGET _Halide_gengen)
+
+        # add_executable requires at least one source file for some
+        # configs (e.g. Xcode), because, uh, reasons, so we'll create
+        # an empty one here to satisfy it
+        set(empty "${CMAKE_CURRENT_BINARY_DIR}/_Halide_gengen.empty.cpp")
+        file(WRITE "${empty}" "/* nothing */\n")
```

Comment:
This needs to be in an `if (NOT EXISTS "${empty}")` to avoid redundant rebuilds on every reconfigure.

Reply:
PTAL